### DISCUSSION TITLE
How to use jwt token from separate backend

### DISCUSSION THREAD

**Discussion by ShourovRoy** [remix-run/remix#1274] [Q&A] [answered]
How can I set token in cookies which comes from separate backend. and how to get that token from cookies in loader function to fetch protected data using that token. Please help me

**Comment by sergiodxa** [comment, marked as answer]
[Create a cookie](https://remix.run/docs/en/v1/api/remix#createcookie) or [session](https://remix.run/docs/en/v1/api/remix#sessions) and store it there, then use that to retrieve the JWT in the loader or action.

**Reply by juniorforlife** [reply]
@sergiodxa From what I understand, the reason we use JWT is to keep the server stateless, meaning the servers don't need to store sessions so it's easier to scale horizontally. With that said, I guess cookie is more appropriate in this case?

**Reply by sergiodxa** [reply]
Sessions still work if you store the session data in the cookie, when using createCookieSessionStorage that's how it works.

You can also scale your server horizontally without issues if the session data is not stored in-memory or in the file system, for example store the session in Redis with a custom createSessionStorage.

**Reply by juniorforlife** [reply]
where is the session stored by default when we use `createCookieSessionStorage`?

**Reply by juniorforlife** [reply]
In addition, let's say I only want to use cookie and no session. How do I delete the cookie when logging out? I don't see any methods like destroyCookie in the documentation.

**Reply by sergiodxa** [reply]
> where is the session stored by default when we use `createCookieSessionStorage`?

In a cookie, all sessions uses cookies, the createCookieSessionStorage stores the session data in the cookie itself, other session storage implementations use the cookie to store a session ID and keep the data somewhere else (the where depends on the implementation).

> In addition, let's say I only want to use cookie and no session. How do I delete the cookie when logging out? I don't see any methods like destroyCookie in the documentation.

There's no way to delete a cookie in HTTP, what you do is to set the cookie again with an expired data, that will make the browser stop sending it and delete it

**Reply by juniorforlife** [reply]
thank you! one last question: let's say I wanna have cookie for showing a Terms & Conditions popup and auth. Should I stuff them both in one cookie or should I separate them to multiple ones?

**Reply by sergiodxa** [reply]
I would probably store the acceptance of Terms & Conditions in the database itself because you may need to have a record to ensure the user accepted them.

But you could use the session too, so you can destroy both things together.

**Reply by lestersconyers** [reply]
From what I can tell `createCookieStorage` and `createCookie` expect an object and serializes/deserializes for you. Unfortunately, stashing access and refresh tokens in cookies with these apis can easily put you over the 4kb size limit for cookies because of the serialization. My team ended up creating a custom `createCookie` api that stores the string as is with no serialization.

**Reply by spacepolice10** [reply]
Probably, that's why I always have unpredictable and inadequately cut tokens if I use `createCookie`

**Comment by spacepolice10** [comment]
> [Create a cookie](https://remix.run/docs/en/v1/api/remix#createcookie) or [session](https://remix.run/docs/en/v1/api/remix#sessions) and store it there, then use that to retrieve the JWT in the loader or action.

Hi, unfortunately, `createCookie` and its `serialize` method breaks JWT and makes it impossible to utilize such tools
